### PR TITLE
*: clean up the deserted syntax of START TRANSACTION READ ONLY

### LINKED ISSUE #24291: Replace the `WITH Staleness` to `AS OF` clause
## Background

This is a subtask of #21094. Ref https://github.com/pingcap/tidb/issues/22765, https://github.com/pingcap/tidb/pull/22506

pingcap/parser#1206 has implemented the syntax of `AS OF` clause, so we can replace begin a staleness read-only transaction with `AS OF` clause, which like:
```
START TRANSACTION READ ONLY AS OF TIMESTAMP '2020-02-02'
```
Note: The `AS OF TIMESTAMP` clause can be used only when the transaction is read-only. If the transaction contains any writes, or if the READ WRITE mode is specified, an error will be returned.
## Implementation

Now that `BeginStmt` is marked with `ReadOnly`, we need to add `AsOfClause` in the statement.  and then replace the origin process logic in https://github.com/pingcap/tidb/pull/21897. and also we may need to implement the `funcCallExpr`, because we should judge the [read_ts_in(xxxx,xxx)](https://github.com/pingcap/tidb/issues/22765#issuecomment-823880049). and should be noted the function is **deterministic** function.

- [x] add the syntax in the `BeginStmt`
- [ ] replace the `with staleness` to `AS OF` cluase

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: close #24291.

Problem Summary: After supporting the `AS OF TIMESTAMP` syntax for `START TRANSACTION READ ONLY` statement, we need to clean up the remaining deserted syntax.

### What is changed and how it works?

* Remove `sessionctx.StalenessTxnOption` since we don't need it anymore.
* Change `NewTxnWithStalenessOption` to `NewTxnWithStartTS`.
* Remove the useless fields from `StartTSOption`.
* Remove the unnecessary test cases.
* Update the test cases.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- `No release note`.

### CODE REVIEW

File: executor/simple.go [hunk 1]
```diff
@@ -75,8 +72,8 @@ type SimpleExec struct {
 	done         bool
 	is           infoschema.InfoSchema
 
-	// StalenessTxnOption is used to execute the staleness txn during a read-only begin statement.
-	StalenessTxnOption *sessionctx.StalenessTxnOption
+	// startTS is used to execute the staleness txn during a read-only begin statement.
+	startTS uint64
```
Comment (Yisaer) [inline executor/simple.go:76]: The name `startTS` is misleading. I prefer `staleTxnStarTS *uint64`

File: store/tikv/tests/extract_start_ts_test.go [hunk 2]
```diff
@@ -53,56 +53,24 @@ func (s *extractStartTsSuite) SetUpTest(c *C) {
 	s.store = probe.KVStore
 }
 
-func (s *extractStartTsSuite) TestExtractStartTs(c *C) {
+func (s *extractStartTsSuite) TestExtractStartTS(c *C) {
```
Comment (Yisaer) [inline store/tikv/tests/extract_start_ts_test.go:56]: The logic is simple now, we can direct remove this test.

**Review by Yisaer — COMMENTED** [review]
Rest LGTM

**Comment by JmPotato** [comment]
CI tests will pass after pingcap/br#1139 is merged.

**Review by Yisaer — APPROVED** [review]
/lgtm

**Pull request merged**